sync2jira/upstream_issue.py

```diff
@@ -311,30 +311,37 @@ def github_issues(upstream, config):
         if issue.get('milestone', None):
             issue['milestone'] = issue['milestone']['title']
 
-        issue['storypoints'] = None
-        issue['priority'] = ''
-        orgname, reponame = upstream.rsplit('/', 1)
-        issuenumber = issue['number']
-        github_project_fields = config['sync2jira']['map']['github'][upstream]['github_project_fields']
-        variables = {"orgname": orgname, "reponame": reponame, "issuenumber": issuenumber}
-        response = get_github_project_details(graphqlurl, headers, ghquery, variables)
-        if response.status_code != 200:
-            log.debug("Error fetching from GitHub: %s", response.text)
-            continue
-        data = response.json()
-        try:
-            nodes = data['data']['repository']['issue']['projectItems']['nodes'][0]['fieldValues']['nodes']
-            for item in nodes:
-                if 'fieldName' in item:
-                    gh_field_name = item.get('fieldName', {}).get('name')
-                    if 'priority' in github_project_fields and gh_field_name in github_project_fields['priority']['fieldmap']:
-                        issue['priority'] = item.get('name')
-                    if 'storypoints' in github_project_fields and gh_field_name in github_project_fields['storypoints']['fieldmap']:
-                        issue['storypoints'] = int(item.get('number'))
-        except (TypeError, KeyError) as err:
-            log.debug("Error fetching %s!r from GitHub %s/%s#%s: %s",
-                      gh_field_name, orgname, reponame, issuenumber, err)
-            continue
+        issue_updates = config['sync2jira']['map']['github'][upstream].get('issue_updates', {})
+        if 'github_project_fields' in issue_updates:
+            issue['storypoints'] = None
+            issue['priority'] = ''
+            orgname, reponame = upstream.rsplit('/', 1)
+            issuenumber = issue['number']
+            github_project_fields = config['sync2jira']['map']['github'][upstream]['github_project_fields']
+            variables = {"orgname": orgname, "reponame": reponame, "issuenumber": issuenumber}
+            response = requests.post(graphqlurl, headers=headers, json={"query": ghquery, "variables": variables})
+            if response.status_code != 200:
+                log.debug("Error fetching from GitHub: %s", response.text)
+                continue
+            data = response.json()
+            gh_issue = data['data']['repository']['issue']
+            if not gh_issue:
+                log.debug("Error fetching issue from GitHub: %s", response.text)
```

Comment:
The `data` key will be always present, if the status code is `200` (see my previous comment for a typical error).
Also, it would be possible to access `issue` like this: `data['data']['repository']['issue']`

So I would prefer to keep this code like this.

Reply:
> The `data` key will be always present, if the status code is `200`

A tiny amount of testing seems to confirm this.

> it would be possible to access `issue` like this: `data['data']['repository']['issue']`

Actually, if the repository is not found by the query, the `data['data']['repository']` will be `null` or `None`, in which case referencing `data['data']['repository']['issue']` will result in a `TypeError`.  However, since, for the moment, Sync2Jira runs as the result of a message from upstream, I think we can assume that the repository will be found.  So, we only need to worry about whether the issue is found (and, again, if we're being driven by a message from upstream, it would be weird if it weren't).

> I would prefer to keep this code like this.

That's fine.

Reply:
We are checking if the value of issue is None. Since we are a using this GraphQL, I think the key `issue` would be always present:
```
...
repository(owner: $orgname, name: $reponame) {
  issue(number: $issuenumber) {
...
```

Reply:
> I think the key `issue` would be always present:

Only if there is a hit on `repository` -- if the repository is not found, then _it's_ value will be `None` and there won't be anywhere for the `issue` key to be stored in the response.  But, as I said, since this code is triggered by a message which originates from the repository, I don't think we need to worry about the repository disappearing before we can issue the query.  🙂

> We are checking if the value of issue is `None`.

Not to be pedantic, but sometimes the difference matters:  we checking if the value of `gh_issue` is "false-y".  It is correct that `None` is a false-y value, but so is `{}` (along with `[]` and `''`), and there could be a big difference between `None` and `{}`.  🥴